ts/main.ts

```diff
@@ -147,6 +161,92 @@ if (!$monsterPage) throw new Error('Could not load Monster Page');
   $monSelect.classList.add('hidden');
+  $monFav.classList.add('hidden');
+  $favoriteTab?.classList.remove('selected');
+  $monsterPage.classList.add('selected');
+  $favTitle?.classList.add('hidden');
+  $monTitle?.classList.remove('hidden');
+  $filter.classList.remove('hidden');
+  $star.classList.add('hidden');
+  $star?.classList.remove('favorite');
+});
+
+if (!$estrella) throw new Error('could not load favorite star click');
+$estrella.addEventListener('click', function () {
+  if ($star.classList[1] === 'favorite') {
+    removeFav();
+  } else if ($star.classList[1] !== 'favorite') {
+    favMon();
+  }
+});
+
+function removeFav(): void {
+  const $card = document.querySelector('.monster-card');
+  if (!$nameMon) throw new Error('Could not load name');
+  if (!$card) throw new Error('Could not load name');
+  if (!$monFavCon?.firstChild)
+    throw new Error('could not load the first child element');
+  if (!$nameMon.textContent) throw new Error('Could not load name');
+  $star?.classList.remove('favorite');
+  for (let i = 0; i < data.favorite.length; i++) {
+    if (Number($nameMon.id) === data.favorite[i].id) {
+      data.favorite.splice(i, 1);
+    }
+  }
+  writeData();
+  if (!$monFav) throw new Error('could not load monster fav page');
+  const $cardToRemove = $monFav.querySelector(
+    `[data-monster-id='${Number($nameMon.id)}']`,
+  );
+  $cardToRemove?.remove();
+}
+
+function favMon(): void {
+  if (!$nameMon) throw new Error('Could not load name');
+  if (!$nameMon.textContent) throw new Error('Could not load name');
+  for (let i = 0; i < data.favorite.length; i++) {
+    if ($nameMon.textContent === data.favorite[i].name) {
+      return;
+    }
+  }
+  $star?.classList.add('favorite');
+  data.favorite.push({
+    name: $nameMon.textContent,
+    photoURL: `images/icons/${$nameMon.textContent}.webp`,
+    id: Number($nameMon.id),
+  });
+  writeData();
+  readData();
+
+  loader();
+}
+
+function loader(): void {
+  if (!$monFavCon) throw new Error('Could not load row');
+  const i = data.favorite.length - 1;
+  const $data = renderMonster(data.favorite[i]);
+  $monFavCon.append($data);
+}
+
+if (!$favoriteTab) throw new Error('Could not load Monster Page');
+$favoriteTab.addEventListener('click', function () {
+  $monLine.classList.add('hidden');
+  $monSelect.classList.add('hidden');
+  $monFav.classList.remove('hidden');
+  $favoriteTab?.classList.add('selected');
+  $monsterPage.classList.remove('selected');
+  $favTitle?.classList.remove('hidden');
+  $monTitle?.classList.add('hidden');
   $filter.classList.remove('hidden');
   $star.classList.add('hidden');
+  $star?.classList.remove('favorite');
 });
+
+document.addEventListener('DOMContentLoaded', domTree);
+function domTree(): void {
+  if (!$monFavCon) throw new Error('Could not load row');
+  for (let i = 0; i < data.favorite.length; i++) {
+    const $data = renderMonster(data.favorite[i]);
+    $monFavCon.append($data);
+  }
+}
```

Comment:
If you have this DOM content loaded I would consider also adding in your initial data fetching function in here too.